README.md

```diff
@@ -7,28 +7,66 @@ Certain things (including this README) are currently a work-in-progress.*
 
 ## What's this?
 
-TODO
+This repository contains both IaC (terraform) and runtime code for the CPF Reporter service and web application.
+The purpose of this service is to provide a web application through which users can upload, validate, and manage financial reporting data for projects funded by federal grants made available through the CPF funding program.
+Users download Microsoft Excel templates that are used for data input and initial validation.
+These files are then regularly uploaded to CPF Reporter for validation and long-term management.
+Users can also download reports to submit to the U.S. Treasury Department for financial reporting,
+as well as audit reports to assist in responding to a Treasury audit.
 
 ## Architecture
 
-TODO
+This service consists of a static web site which provides a single-page client application (SPA).
+The client application primarily interacts with the GraphQL API,
+for which requests are handled by an AWS Lambda function backed by a PostgresQL database.
+When users (via the client application) upload files to S3,
+an event-driven workflow is initiated that validates the contents of the upload
+and stores the validation results in the PostgresQL database.
+
+The client web application is hosted in a CloudFront distribution that serves static assets
+built from the RedwoodJS `web`-side source code and uploaded to the distribution's origin bucket.
+Lambda function are built from RedwoodJS `api`-side and Python source code,
+which comprise the server-side compute components of the service.
+
+<details>
+  <summary>Component-level architecture</summary>
+  <img src="docs/component-level-architecture.png"/>
+</details>
+
 
 ## Code Organization
 
-Code for this service can generally be considered under three categories:
+Code for this service can generally be considered under four categories:
+
+1. Web-side TypeScript code, located under `web/`, which is used to build the single-page application (SPA) that runs in users' browsers.
+2. API-side TypeScript code, located under `api/`, which powers the GraphQL API consumed by the SPA.
```

Comment:
Is it worth breaking this down further by explaining:
2. API-Side Typescript code, located under `api/` which has two entrypoints:
  - GraphQL entrypoint - this begins at `api/src/functions/graphql.ts` and powers the GraphQL API consumed by the SPA
  - One-off endpoints - this begins at `api/src/functions/{functionName}/{functionName.ts}` and powers one-off lambda endpoints that can be consumed by the SPA and other clients.